### PR TITLE
Allow setting permissions on mountpoints

### LINKED ISSUE #443: Unable to set permission when mounting additional partitions, misleading example
It is not possible to set permission when mounting additional partitions using manifest file http://bootstrap-vz.readthedocs.io/en/master/manifests/index.html?highlight=mount#volume - mount point always has permission 0755. 

I'm not saying it is a high priority issue (we use single partition images), but it makes example manifest file https://github.com/andsens/bootstrap-vz/blob/master/manifests/examples/kvm/stretch-virtio-partitions.yml misleading - ```/tmp``` and ```/var/tmp``` should not have 0755 permissions but 1777.

### PR DESCRIPTION
**Pull request by vmlintu-nosto**
Add new attribute 'mode' for partitions to set the permissions of the
mountpoints. Without this having /tmp on separate partition results
in error as the permissions of /tmp are 0755. Setting them to 01777
fixes the setup.

Fixes #443

### CODE REVIEW

**Comment by andsens** [comment]
Nice. I'll have to review this more in-depth before I can merge. Your fix also accounts for nested mounts, I like it.

p.s.: You have some errors in travis that need to be fixed.

**Comment by andsens** [comment]
Oh wait. Actually one could just wait until everything is mounted and then run a single task, which chmods all the mountpoints. This way you don't need to touch all that code.

I'd add a task to `common.tasks.filesystem` and define MountAdditional as a predecessor: https://github.com/andsens/bootstrap-vz/blob/25ec499868e54eeab48e5c10940bdfcd570461d0/bootstrapvz/common/tasks/filesystem.py#L102-L108
Then you just need to add the new task to the mounting group: https://github.com/andsens/bootstrap-vz/blob/25ec499868e54eeab48e5c10940bdfcd570461d0/bootstrapvz/common/task_groups.py#L75-L83

**Comment by vmlintu-nosto** [comment]
Thanks for the pointers. I did a new version of the patch that adds a new task to test it out. Is there a better way of getting the mode attribute from the manifest in the task?

File: bootstrapvz/base/manifest-schema.yml [hunk 1]
```diff
@@ -191,6 +191,7 @@ definitions:
         minItems: 1
         type: array
       size: {$ref: '#/definitions/bytes'}
+      mode: {type: integer}
```
[the remark this answers is not in the document]
Comment (vmlintu-nosto) [reply bootstrapvz/base/manifest-schema.yml]: Using 01777 worked on python 2.7 for octal conversions, but may break with other python versions, so I changed this now to string. I pushed a new squashed version of the PR.

**Comment by andsens** [comment]
Looks great! Merging...

**Pull request merged**